docs/DeveloperGuide.md

```diff
@@ -694,8 +845,217 @@ Use case ends.
 * 2a. The list is empty.
 
    Use case ends.
+   
+</div>
+
+<div markdown="block" class="alert alert-info">
+
+**Use case: UC07 - Add a session**
 
-**Use case: UC06 - Exit ATAS**
+**Guarantees :**
+* Session will be added only if the input session name and session date are valid.
+* The session list will not have duplicate sessions after adding.
+
+**MSS**
+
+1.  User requests to add a session with a session name and date.
+1.  ATAS checks if the inputs are valid and scans though the session list for duplication.
+1.  ATAS adds the session to the session list.
+1.  ATAS sorts the session list according to date.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The given date is invalid.
+  * 2a1. ATAS shows an error message.
+  
+  Use case ends.
+  
+* 2b. A session in the session list has the same session name.
+  * 2b1. ATAS shows an error message.
+  
+  Use case ends.
+  
+</div>
+
+<div markdown="block" class="alert alert-info">
+
+**Use case: UC08 - Delete a session**
+
+**MSS**
+
+1.  User requests to delete a session with the given index.
+1.  ATAS asks for confirmation.
+1.  User confirms.
+1.  ATAS deletes the session.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given session index is invalid.
+  * 1a1. ATAS shows an error message.
+  
+  Use case ends.
+
+* 2a. User denies confirmation.
+
+  Use case ends.
+  
+</div>
+
+<div markdown="block" class="alert alert-info">
+
+**Use case: UC09 - Edit a session**
+
+**Guarantees :**
+* Session will be edited only if the edited session name and session date are valid.
+* The session list will not have duplicate sessions after edit.
+
+**MSS**
+
+1.  User requests to edit a session with session name and date with the session index of that session.
+1.  ATAS checks if the input is valid and scans through the session list for duplication.
+1.  ATAS asks for confirmation.
+1.  User confirms.
+1.  ATAS edits the session
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The given session index is invalid
+   * 2a1. ATAS shows an error message.
+   
+   Use case ends.
+
+* 2b. The session date given is invalid
+  * 2b1. ATAS shows an error message.
+  
+  Use case ends.
+
+* 3a. User denies confirmation.
+
+  Use case ends.
+
+</div>
+
+<div markdown="block" class="alert alert-info">
+
+**Use case: UC10 - Clear the session list**
+
+**MSS**
+
+1.  User requests to clear the session list.
+1.  ATAS asks for confirmation.
+1.  User confirms.
+1.  ATAS clears the session list.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The session list is empty
+   * 1a1. ATAS shows an empty session list.
+   
+   Use case ends.
+   
+* 2a. User denies confirmation.
+
+  Use case ends.
+   
+</div>
+
+<div markdown="block" class="alert alert-info">
+
+**Use case: UC11 - Enter a session**
+
+**MSS**
+
+1.  User requests to enter a session with a session index.
+1.  ATAS shows the attributes of that session.
+
+   Use case ends.
+
+**Extensions**
```

Comment:
Perhaps another extension could be added to highlight what happens if the user tries to enter the same session he is already in?